core/src/test/java/com/linecorp/armeria/server/ServiceRequestContextTest.java

```diff
@@ -69,16 +82,86 @@ void mapCurrent() {
         assertThat(ServiceRequestContext.mapCurrent(ctx -> "foo", () -> "bar")).isEqualTo("bar");
         assertThat(ServiceRequestContext.mapCurrent(Function.identity(), null)).isNull();
 
-        final ServiceRequestContext ctx = ServiceRequestContext.of(req);
+        final ServiceRequestContext ctx = serviceRequestContext();
         try (SafeCloseable unused = ctx.push()) {
+            assertThat(onEnterExitStack.size()).isOne();
             assertThat(ServiceRequestContext.mapCurrent(c -> "foo", () -> "bar")).isEqualTo("foo");
             assertThat(ServiceRequestContext.mapCurrent(Function.identity(), null)).isSameAs(ctx);
         }
+        assertThat(onEnterExitStack.size()).isZero();
 
-        try (SafeCloseable unused = ClientRequestContext.of(req).push()) {
+        try (SafeCloseable unused = clientRequestContext().push()) {
             assertThatThrownBy(() -> ServiceRequestContext.mapCurrent(c -> "foo", () -> "bar"))
                     .isInstanceOf(IllegalStateException.class)
                     .hasMessageContaining("not a server-side context");
         }
     }
+
+    @Test
+    void pushReentrance() {
```

Comment:
Ditto - would be nice to have more assertions for all new test cases.

Reply:
Found a bug after adding more test cases. Thanks!